thinc/layers/with_array.py

```diff
@@ -2,42 +2,44 @@
 
 from ..model import Model
 from ..config import registry
-from ..types import Array2d, Floats2d, Padded, Ragged, ArrayXd, Floats3d
-from ..types import List2d
+from ..types import Padded, Ragged, ArrayXd, Array3d, ListXd
 
-
-SeqT = TypeVar("SeqT", bound=Union[Padded, Ragged, List2d, ArrayXd])
+ArrayTXd = TypeVar("ArrayTXd", bound=ArrayXd)
+SeqT = TypeVar("SeqT", bound=Union[Padded, Ragged, ListXd, ArrayXd])
 
 
 @registry.layers("with_array.v1")
-def with_array(layer: Model[ArrayXd, ArrayXd], pad: int = 0) -> Model[SeqT, SeqT]:
+def with_array(layer: Model[ArrayTXd, ArrayTXd], pad: int = 0) -> Model[SeqT, SeqT]:
     """Transform sequence data into a contiguous 2d array on the way into and
     out of a model. Handles a variety of sequence types: lists, padded and ragged.
     If the input is a 2d array, it is passed through unchanged.
     """
-    return Model(
+    model: Model[SeqT, SeqT] = Model(
         f"with_array({layer.name})",
         forward,
         init=init,
         layers=[layer],
         attrs={"pad": pad},
         dims={name: layer.maybe_get_dim(name) for name in layer.dim_names},
     )
+    return model
 
 
-def forward(model: Model[SeqT, SeqT], Xseq: SeqT, is_train: bool):
+def forward(
```

Comment:
I'd rewrite this with shorter variable names as before, and keep the `return` statements within each branch